### ISSUE TITLE
Ethernet support? ESP32 port?

### ISSUE THREAD

**Issue by DjordjeMandic** [omersiar/esp-rfid#427]
Did someone port this to esp32 with ethernet? Is it possible to add ethernet to this project? It would suit it more, poe + ethernet perfect combination. Wifi comm. can be easily jammed.

**Comment by evazzoler** [comment]
I agree! And ethernet is more stable, almost a must when you use MQTT in a full 2.4 GHz band like when you are in a building in the city center...

**Comment by evazzoler** [comment]
Please, add the possibilty to disable wireless and connect a cheap W5100 interface to the board.

**Comment by sinclairfr** [comment]
I support this request. Ethernet would be so much more stable. Also, do you guys know how hard (or how easy) it would be to port this project to a raspberry pi ?  Thanks

**Comment by mkrasselt1** [comment]
I am currently in the works to port this to ttgo esp32-poe board  and build a custom pcb head for it

https://github.com/mkrasselt1/esp-rfid

**Comment by matjack1** [comment]
That's great @mkrasselt1! Are you planning to keep also the ESP8266 compatibility? If you would like to contribute back to this repo I'm happy to review your code.

**Comment by mkrasselt1** [comment]
> Are you planning to keep also the ESP8266 compatibility?
If by coincidence I own the right Chip to test it against, I could do it but it would be a different software.  Which chip are you planing to use?

**Comment by matjack1** [comment]
@mkrasselt1 I have a NodeMCU and a Wemos D1 at hand, but I think they are all very very similar.

**Comment by mkrasselt1** [comment]
No, I mean which Ethernet module are you planning to use?

**Comment by matjack1** [comment]
Ah, no I have no idea about ethernet modules yet :) If you have recommendations that would be interesting

**Comment by t0bse** [comment]
any news on this?

**Comment by DjordjeMandic** [comment]
> Ah, no I have no idea about ethernet modules yet :) If you have recommendations that would be interesting

You can use either the SPI ones like W5500 W5100 ENC28J60 with ESP8266 or utilize hardware eth mac of ESP32 and just add eth phy module like LAN8720. 

Keep in mind that LAN8720 is just physical layer that can be used with ESP32's intergrated ethernet mac.

For ESP8266 you must use SPI modules like W5500 W5100 ENC28J60

**Comment by mkrasselt1** [comment]
So for all of you, yes I am still working on it - but I plan to update the PCB as well :))

**Comment by DjordjeMandic** [comment]
Any progress by anyone?

**Comment by M4rkoV2** [comment]
@DjordjeMandic did you manage to do it somehow?

**Comment by DjordjeMandic** [comment]
Nope, didnt even try yet, i was looking into other solutions with home assistant. Might do it one day but i dont have ESP32 eth dev enviroment (ethernet hardware)

**Comment by mkrasselt1** [comment]
for now, i am using esphome to achive my goals - so far i wont be of much help to this project :/ cant say whenever i will have time reason to look into this as well
